File: ios/Classes/Bridge/SetGPPConsentBridge.swift
import Usercentrics

struct SetGPPConsentBridge : MethodBridge {

    let name: String = "setGPPConsent"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        let argsDict = call.arguments as! Dictionary<String, Any>
        let sectionName = argsDict["sectionName"] as! String
        let fieldName = argsDict["fieldName"] as! String
        let value = argsDict["value"] ?? NSNull()
        usercentrics.shared.setGPPConsent(sectionName: sectionName, fieldName: fieldName, value: value)
        result(nil)
    }
}

File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/SetGPPConsentBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton

internal class SetGPPConsentBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    override val name: String
        get() = "setGPPConsent"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        val argsMap = call.arguments as Map<*, *>
        val sectionName = argsMap["sectionName"] as String
        val fieldName = argsMap["fieldName"] as String
        val value = normalizeValue(argsMap["value"])
        usercentrics.instance.setGPPConsent(sectionName, fieldName, value)
        result.success(null)
    }

    private fun normalizeValue(value: Any?): Any {
        if (value == null) return Unit
        if (value is Double && value % 1.0 == 0.0) return value.toInt()
        return value
    }
}
